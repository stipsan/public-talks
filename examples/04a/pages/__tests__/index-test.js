import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import App from '../index'

describe('With Enzyme', () => {
  it('App renders correctly', () => {
    const app = mount(<App />)
    expect(toJson(app, { mode: 'deep' })).toMatchSnapshot()

    app.setProps({ isDark: true })
    expect(toJson(app, { mode: 'deep' })).toMatchSnapshot()
  })
})
