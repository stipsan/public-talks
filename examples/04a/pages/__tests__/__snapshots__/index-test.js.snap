// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Enzyme App renders correctly 1`] = `
.c0 {
  background: palevioletred;
  color: white;
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
}

.c1 {
  background: white;
  color: palevioletred;
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
}

<div>
  Theme is 
  Light
  <button
    className="c0"
  />
  <button
    className="c1"
  />
</div>
`;

exports[`With Enzyme App renders correctly 2`] = `
.c1 {
  background: palevioletred;
  color: white;
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
}

.c3 {
  background: white;
  color: palevioletred;
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
}

.c0 {
  background: crimson;
  color: black;
  border-color: crimson;
}

.c2 {
  background: black;
  color: crimson;
  border-color: crimson;
}

<div>
  Theme is 
  Dark
  <button
    className="c0 c1"
  />
  <button
    className="c2 c3"
  />
</div>
`;
